lib/routers/profile/pil.js

```diff
@@ -120,11 +129,13 @@ router.put('/:pil/:action',
     if (action === 'revoke') {
       return whitelist('comments')(req, res, next);
     }
-    whitelist('procedures', 'notesCatD', 'notesCatF', 'species')(req, res, next);
+    whitelist('procedures', 'notesCatD', 'notesCatF', 'species', 'establishment')(req, res, next);
   },
   validate,
   updateDataAndStatus(),
-  submit()
+  (req, res, next) => {
+    return submit(req.params.action)(req, res, next);
+  }
```

Comment:
This seems like an unnecessary dance. You can do `const action = req.params.action` inside the `submit` function and leave this code as it was.